lib/streamlit/runtime/fragment.py

```diff
@@ -79,3 +84,30 @@ def delete(self, key: str) -> None:
 
     def clear(self) -> None:
         self._fragments.clear()
+
+
+@overload
+def fragment(
+    func: F,
+    *,
+    run_every: float | None = None,
+) -> F:
+    ...
+
+
+@overload
+def fragment(
+    func: None = None,
+    *,
+    run_every: float | None = None,
+) -> Callable[[F], F]:
+    ...
+
+
+@gather_metrics("experimental_fragment")
+def fragment(  # type: ignore[empty-body]
+    func: F | None = None,
+    *,
+    run_every: float | None = None,
+) -> Callable[[F], F] | F:
+    pass
```

Comment:
Leaving the decorator as a stub to be implemented in a subsequent PR

Reply:
Would it make sense to leave a short comment for each overload explaining for which scenario it is meant? For example, the second overload it's not directly clear to me what `def fragment` is doing when `func` is `None`. 
If this is a common Python pattern, ignore it and I will just read about it 😁

Reply:
I don't think this type of thing is too uncommon (it's to support being able to write `@fragment(run_every=5.0)`), but I couldn't find a place in our codebase that does this, so I'll add a comment for the second overload.